client/components/ShareButton.js

```diff
@@ -17,16 +19,34 @@ const useStyles = ((canShare) => makeStyles((theme) => ({
 })));
 
 export default () => {
+  const [state, setState] = useState({showSnackbar: false});
   const shareData = {url: window.location.href};
   const classes = useStyles(navigator.canShare && navigator.canShare(shareData))();
-  // TODO: handle navigator.share() failures.
-  return <Button
-      variant='contained'
-      color='primary'
-      startIcon={<ShareIcon />}
-      className={classes.button}
-      onClick={() => navigator.share(shareData)}
-    >
-      Share
-    </Button>
+
+  const share = (shareData) => {
+    navigator.share(shareData).catch((e) => {
```

Comment:
nit: use an async lambda?
```js
const share = async (shareData) => {
  try {
    await navigator.share(shareData);
  } catch (e) {
    // ...
  }
};
```

Reply:
Done, but why is this preferred?

Reply:
In case we were to add more logic after a share that might be async too I feel like it's easier to read with async / await.